httpcore/_async/http11.py

```diff
@@ -123,7 +123,7 @@ def mark_as_ready(self) -> None:
             event = await self._receive_event(timeout)
             if isinstance(event, h11.Data):
                 yield bytes(event.data)
-            elif isinstance(event, h11.EndOfMessage):
+            elif isinstance(event, (h11.EndOfMessage, h11.PAUSED)):
```

Comment:
From the [h11 docs](https://h11.readthedocs.io/en/latest/api.html#flow-control):

> PAUSED is a little more subtle: [...] A successful CONNECT or Upgrade: request has caused the connection to switch to some other protocol (see Switching protocols). This PAUSED state is permanent; you should abandon this Connection and go do whatever it is you’re going to do with your new protocol.

This was required to be able to consume the proxy's response body without closing the socket.